lib/rest2.js

```diff
@@ -98,6 +100,7 @@ class RESTv2 {
     this._transform = !!opts.transform
     this._agent = opts.agent
     this._affCode = opts.affCode
+    this._timeout = opts.timeout
```

Comment:
hmm what if user provides {}, in that case this._timeout would be unefined and nodejs as fallback uses timeout 0 which is infinite, I think here we should do this:
```js
const _isNil = require('lodash/isNil')
...
this._timeout = _isNil(opts.timeout) ? BASE_TIMEOUT : opts.timeout
```

the reason why not to do simply || BASE_TIMEOUT is because it won't allow 0

Reply:
for user `timeout` is documented as number
in this case, setting `timeout` to a different data type is the responsibility of the user
with an inappropriate data type, instead of setting a default value, it would be better to throw an exception. what do you think?

Reply:
@vigan-abd

Reply:
yep we could also throw the error, so if(!_isNil && !_isInteger) should throw
but please be aware of 0 input